pkg/runtime/local/local.go

```diff
@@ -55,31 +55,18 @@ func (r *Runtime) RunGadget(gadgetCtx runtime.GadgetContext) (out []byte, err er
 		return out, fmt.Errorf("instantiating gadget: %w", err)
 	}
 
-	err = gadgetInstance.Init(gadgetCtx)
-	if err != nil {
-		return out, fmt.Errorf("initializing gadget: %w", err)
-	}
-
-	// Deferring getting results and closing to make sure operators got their chance to clean up properly beforehand
-	defer func() {
-		if closer, ok := gadgetInstance.(gadgets.CloseGadget); ok {
-			log.Debugf("calling gadget.Close()")
-			closer.Close()
-		}
-
-		// No need to get results if gadget failed
+	// Initialize gadgets, if needed
+	if initRunClose, ok := gadgetInstance.(gadgets.InitRunClose); ok {
+		log.Debugf("calling gadget.Init()")
```

Comment:
We could add more debug messages in the runtime to clarify things that don't happen, like "gadget.Init() not present, skipping" (in addition to the messages we already print), to better visualize the lifecycle of the gadget.